src/timeline/Timeline.js

```diff
@@ -0,0 +1,216 @@
+// const RIGHT_MARGIN = 10
+// const CALENDER_HEIGHT = 2400;
+// const EVENT_TITLE_HEIGHT = 15
+const TEXT_LINE_HEIGHT = 17;
+// const MIN_EVENT_TITLE_WIDTH = 20
+// const EVENT_PADDING_LEFT = 4
+
+function range(from, to) {
+  return Array.from(Array(to), (_, i) => from + i);
+}
+
+let {width: dimensionWidth} = Dimensions.get('window');
+
+export default class Timeline extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    const {start = 0, end = 24} = this.props;
+    this.styles = styleConstructor(props.styles, (end - start) * 100);
+    this.calendarHeight = (end - start) * 100;
+    const width = dimensionWidth - LEFT_MARGIN;
+    const packedEvents = populateEvents(props.events, width, start);
+    let initPosition =
+      _.min(_.map(packedEvents, 'top')) - this.calendarHeight / (end - start);
+    initPosition = initPosition < 0 ? 0 : initPosition;
+    this.state = {
+      _scrollY: initPosition,
+      packedEvents,
+    };
+  }
+
+  componentWillReceiveProps({events, start = 0}) {
+    const width = dimensionWidth - LEFT_MARGIN;
+    this.setState({
+      packedEvents: populateEvents(events, width, start),
+    });
+  }
+
+  componentDidMount() {
+    this.props.scrollToFirst && this.scrollToFirst();
+  }
+
+  scrollToFirst() {
+    setTimeout(() => {
+      if (this.state && this.state._scrollY && this._scrollView) {
+        this._scrollView.scrollTo({
+          x: 0,
+          y: this.state._scrollY,
+          animated: true,
+        });
+      }
+    }, 1);
+  }
+
+  // _renderRedLine() {
+  //   const offset = 100;
+  //   const {start = 0} = this.props;
+  //   const timeNowHour = moment().hour();
+  //   const timeNowMin = moment().minutes();
+  //   return (
+  //     <View
+  //       key={'timeNow'}
+  //       style={[
+  //         this.styles.lineNow,
+  //         {
+  //           top: offset * (timeNowHour - start) + (offset * timeNowMin) / 60,
+  //           width: dimensionWidth - 20,
+  //         },
+  //       ]}
+  //     />
+  //   );
+  // }
+
+  _renderLines() {
+    const {format24h, start = 0, end = 24} = this.props;
+    const offset = this.calendarHeight / (end - start);
+
+    return range(start, end + 1).map((i, index) => {
+      let timeText;
+      if (i === start) {
+        timeText = '';
+      } else if (i < 12) {
+        timeText = !format24h ? `${i} AM` : i;
+      } else if (i === 12) {
+        timeText = !format24h ? `${i} PM` : i;
+      } else if (i === 24) {
+        timeText = !format24h ? '12 AM' : 0;
+      } else {
+        timeText = !format24h ? `${i - 12} PM` : i;
+      }
+      return [
+        <Text
+          key={`timeLabel${i}`}
+          style={[this.styles.timeLabel, {top: offset * index - 6}]}>
+          {timeText}
+        </Text>,
+        i === start ? null : (
+          <View
+            key={`line${i}`}
+            style={[
+              this.styles.line,
+              {top: offset * index, width: dimensionWidth - 20},
```

Comment:
The `20` here and on line #124 should be in a constant so we'll understand its meaning and avoid repetition.

Reply:
@Inbal-Tish Fixed!